src/Entity/SmsMessage.php

```diff
@@ -135,26 +142,31 @@ public function removeOption($name) {
    * {@inheritdoc}
    */
   public function getResult() {
+    // Check the temporary store first as that contains the most recent value.
+    // Also, if the entity is new return that value (can be null).
+    if ($this->result || $this->isNew()) {
+      return $this->result;
+    }
     $results = $this->entityTypeManager()
       ->getStorage('sms_result')
-      ->loadByProperties(['parent' => $this->id()]);
-    return $results ? reset($results) : NULL;
+      ->loadByProperties(['sms_message' => $this->id()]);
+    $this->result = $results ? reset($results) : NULL;
+    return $this->result;
   }
 
   /**
    * {@inheritdoc}
    */
   public function setResult(StdMessageResultInterface $result = NULL) {
-    if ($result === NULL) {
-      // Find any pre-existing report and remove the parent-child relationship.
-      if ($previous_result = $this->getResult()) {
-        // @todo.
-        $previous_result->setParent(NULL);
-      }
+    // Throw an exception if there is already a result for this SMS message.
+    $previous_result = $this->getResult();
+    if ($previous_result) {
+      throw new \InvalidArgumentException('SMS message result cannot be changed or updated.');
```

Comment:
Even though that was before I came up with the temporary storage for `setResult()`, I guess this can go away. I take another look.